include/swift/SIL/ApplySite.h

```diff
@@ -383,6 +383,8 @@ class ApplySite {
   static bool classof(const SILInstruction *inst) {
     return bool(ApplySiteKind::fromNodeKind(inst->getKind()));
   }
+
+  void dump() const { getInstruction()->dump(); }
```

Comment:
Can you wrap this with a used attribute and an obsolute attribute. E.x.:

```
LLVM_ATTRIBUTE_DEPRECATED(void dump() const, "only for use in the debugger");
```

Reply:
(Forgot to add the LLVM_ATTRIBUTE_USED). So it would actually bt his:

```
LLVM_ATTRIBUTE_DEPRECATED(void dump() const LLVM_ATTRIBUTE_USED, "only for use in the debugger");
```

Reply:
I didn't use LLVM_ATTRIBUTE_DEPRECATED because it's not a declaration, it's a trivial inline function definition around SILInstruction::dump() and I didn't want to declare it separately. Incidentally, SILInstruction::dump() is not declared as LLVM_ATTRIBUTE_DEPRECATED either, and it's expected to call these functions within the source to generate debug output.

LLVM_ATTRIBUTE_USED makes perfect sense.

I frankly don't understand the use-case for using both of these attributes together. What problem is solved by polluting the code with LLVM_ATTRIBUTE_DEPRECATED? Sometimes people (like me) accidentally commit a wayward debug print call that's not under a flag for a short time, but I don't see that as a serious problem that the attribute can solve in the common case.